library/std/src/os/unix/process.rs

```diff
@@ -12,15 +12,20 @@ use crate::sealed::Sealed;
 use crate::sys;
 use crate::sys_common::{AsInner, AsInnerMut, FromInner, IntoInner};
 
-#[cfg(not(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon")))]
-type UserId = u32;
-#[cfg(not(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon")))]
-type GroupId = u32;
-
-#[cfg(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon"))]
-type UserId = u16;
-#[cfg(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon"))]
-type GroupId = u16;
+use cfg_if::cfg_if;
+
+cfg_if! {
+    if #[cfg(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon"))] {
+        type UserId = u16;
+        type GroupId = u16;
+    } else if #[cfg(target_os = "nto")] {
+        type UserId = i32;
+        type GroupId = i32;
```

Comment:
Really? Why i32s?

Reply:
This is not directly visible in the official documentation (at least I've not found it) and unfortunately the toolchain is not open source so I cannot directly link to it. The documentation about [setuid()](https://www.qnx.com/developers/docs/7.1/#com.qnx.doc.neutrino.lib_ref/topic/s/setuid.html) gives the hint to not use negative values for the user ID.
Also, I verified it in the header files:
`sys/types.h` defines `__UID_T`:
```c
typedef __UID_T		uid_t;
```
which is defined in `/sys/target_nto.h`:
```c
#define __UID_T		_INT32
```
`_INT32` is always a signed 32 bit integer. Same applies for group ID and process ID.

Reply:
Huh, can you inline that remark, or some other comment to make it clear that this is morally a `u31`?

Reply:
Addressed with 025471e55e9174d98ed4bae2a86407e600dbdc34.